src/library/searchquery.h

```diff
@@ -90,6 +92,23 @@ class TextFilterNode : public QueryNode {
     QString m_argument;
 };
 
+class CrateFilterNode : public QueryNode {
+  public:
+    CrateFilterNode(const QSqlDatabase& database,
+                    const QString& argument,
+                    const CrateStorage* pCrateStorage);
+
+    bool match(const TrackPointer& pTrack) const override;
+    QString toSql() const override;
+
+  private:
+    QSqlDatabase m_database;
```

Comment:
This member should vanish according to my comment regarding the database connection